SparkleSpin/Controllers/PlayerViewController.swift

```diff
@@ -14,21 +14,54 @@ class PlayerViewController: UIViewController {
     @IBOutlet weak var playerEntryTextField: CustomTextField!
     @IBOutlet weak var addPlayerButton: LightButton! 
     
-    let playerViewModel = PlayerViewModel()
+    @objc let playerViewModel = PlayerViewModel()
     var doneBarButton: UIBarButtonItem?
+    var playerViewModelObservationToken: NSKeyValueObservation?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         setNavigationBarTitle()
         registerNib()
         setupTableView()
         addDoneBarButtonItem()
+        setObservationToken()
     }
     
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        
+        // Invalidate observation tokens to stop observing
+        playerViewModelObservationToken?.invalidate()
```

Comment:
I wonder if invalidating the observer here could lead to some bugs 🤔. Let's say for example that you push another ViewController to the stack, that means this controller is still in memory but you have invalidated the observer and if the user goes back to this controller the observer won't work anymore because it was invalidated when the second controller got pushed. I think one solution could be reading the observer every time on `viewWillAppear` instead of doing it in `viewDidAppear` or doing something like [this](https://www.objc.io/blog/2018/04/24/bindings-with-kvo-and-keypaths/)

Reply:
Thanks! I now set the observer in `viewWillAppear`.